src/render/colorScale.ts

```diff
@@ -328,28 +328,41 @@ const thresholds = (min: number, max: number, n: number): Array<number> =>
   );
 
 export type ColorScaleParams = {
+  customColors: Array<HEXColor>,
   colorScheme: string,
   min: number,
   max: number,
   isReverse?: boolean,
   isContinuous?: boolean
 }
 
-const colorScale = ({colorScheme, min, max, isReverse = false, isContinuous = false}: ColorScaleParams) => {
-  if (isValidColorSchemeName(colorScheme)) {
-    const colors = COLOR_SCHEMES[colorScheme].map(hexToIntColor);
-    const range = isReverse ? colors.reverse() : colors;
+const colorScale = ({ colorScheme, customColors, min, max, isReverse = false, isContinuous = false }: ColorScaleParams) => {
+  let colors: Array<HEXColor>;
 
-    if (isContinuous) {
-      const domain = intervals(min, max, range.length);
-      return scaleLinear(domain, range).clamp(true);
+  if (colorScheme) {
+    if (isValidColorSchemeName(colorScheme)) {
+      colors = COLOR_SCHEMES[colorScheme];
     } else {
-      const domain = thresholds(min, max, range.length);
-      return scaleThreshold(domain, range);
+      throw new Error(`"${colorScheme}" is not a supported color scheme`);
     }
+  } else if (customColors.length >= 3) {
+    colors = customColors;
   } else {
-    throw new Error(`"${colorScheme}" is not a supported color scheme`);
+    throw new Error(`You must provide at least 3 colors`);
```

Comment:
Why at least 3 colors?
I'd say 2 colors is still a legitimate color ramp.
Have you tried with 2 colors?

Reply:
Not sure why I picked 3 as the minimum for a ramp, I've reduced this to 2 now